core/primitives/src/views.rs

```diff
@@ -226,6 +237,13 @@ pub struct StateItem {
     pub proof: Vec<()>,
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct ViewAccessKeyResult {
+    pub access_key: AccessKey,
+    // proof is a serialized array of nodes that make up a merkle path from the key to the root of the shard.
+    pub proof: Option<Vec<Arc<[u8]>>>,
```

Comment:
Why `Arc<[u8]>` here? If it is serialized it sounds like `Vec<u8>` would be more appropriate?